kiwi/bootloader/config/grub2.py

```diff
@@ -227,13 +230,54 @@ def setup_disk_image_config(
                 self.boot_directory_name, 'grub.cfg'
             ]
         )
+
+        if self.validate_use_of_linuxefi:
+            # On systems that uses GRUB_USE_LINUXEFI with grub2 version
+            # less than 2.04 there is no support for dynamic EFI
+            # environment checking. In this condition we extend the grub
+            # setup to add this support. The change kiwi does is as
+            # follows:
+            #
+            # * Apply only on grub < 2.04
+            #    1. Modify 10_linux to set linux/initrd as variables
+            #    2. Write hybrid setup as 01_efihybrid
+            if not CommandCapabilities.check_version(
+                self._get_grub2_mkconfig_tool(),
+                version_waterline='2.04', raise_on_error=False
+            ):
+                linux_config = os.sep.join(
+                    [self.root_mount.mountpoint, 'etc/grub.d/10_linux']
+                )
+                with open(linux_config) as grub_linux_config_file:
+                    grub_linux_config = grub_linux_config_file.read()
+                    grub_linux_config = re.sub(
+                        r'([ \t]+)linux[efi]*([ \t]+)', r'\1\$linux\2',
```

Comment:
Would also match `linuxeieieif`, but shouldn't actually cause an issue AFAICT

Reply:
yes I know but thought it doesn't matter